float8_experimental/float8_dynamic_linear.py

```diff
@@ -72,14 +76,36 @@ def forward(self, x):
         y = cast_to_float8_e5m2_bw(y, self.backward_config)
         return y
 
+    def static_quantize_weight(self, dtype: torch.dtype = torch.float8_e4m3fn) -> None:
+        """Used to perform static_quantization, useful for inferenece where weights are not updated."""
+
+        scale = tensor_to_scale(self.weight, dtype)
+        quantized_weight = to_fp8_no_autograd(
+            self.weight,
+            scale,
+            dtype,
+            self.forward_config,
+        )
+        self.weight = nn.Parameter(quantized_weight)
+
     @classmethod
-    def from_float(cls, mod, emulate: bool = False) -> "Float8DynamicLinear":
+    def from_float(
+        cls,
+        mod,
+        emulate: bool = False,
+        static_quantize_weight: bool = False,
+        activation_scale: Optional[torch.Tensor] = None,
```

Comment:
we would probably want to provide two options:
1. user specifies the scale they want (for example, for cases where the preceding activation has a bounded range)
2. user specifies how to calculate the scale from calibration data

For (2), the current quantization UXs do this with `Observer` objects (https://fburl.com/code/ady6pz23).  cc @jerryzh168 on if torchao has any different plans to do this in future UXs.